src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -884,21 +886,91 @@ void CompilerDirectivesPrintDCmd::execute(DCmdSource source, TRAPS) {
 
 CompilerDirectivesAddDCmd::CompilerDirectivesAddDCmd(outputStream* output, bool heap) :
                            DCmdWithParser(output, heap),
-  _filename("filename","Name of the directives file", "STRING",true) {
+  _filename("filename", "Name of the directives file", "STRING", true),
+  _force_deopt("-d", "Force deoptimization of affected methods.", "BOOLEAN", false, "false") {
```

Comment:
I agree with Paul a generic alternative like `refresh` would be better.

Reply:
Changed the flag, description and names to 'refresh', 'r'.